pkg/services/ngalert/models/alertmanager.go

```diff
@@ -13,7 +15,24 @@ type AlertConfiguration struct {
 
 // GetLatestAlertmanagerConfigurationQuery is the query to get the latest alertmanager configuration.
 type GetLatestAlertmanagerConfigurationQuery struct {
+	Result *AlertConfiguration
+}
+
+// GetAlertmanagerConfigurationQuery is the query to get the latest alertmanager configuration.
+type GetAlertmanagerConfigurationQuery struct {
 	ID int64
 
 	Result *AlertConfiguration
 }
+
+// SaveAlertmanagerConfigurationCmd is the command to save an alertmanager configuration.
+type SaveAlertmanagerConfigurationCmd struct {
+	AlertmanagerConfiguration string
+	ConfigurationVersion      string
+
+	Result *AlertConfiguration
```

Comment:
What's the point of this field, it's not used is it?

In any case, idiomatic Go should be to return the result, instead of via a pointer argument. When using the bus, the `Result` field has a purpose, but since we're moving away from the bus (to drop global state, and use static/type safe instead of dynamic dispatch) there's no real reason to not use normal return values any longer :)